app/src/main/java/com/pvp/app/ui/theme/Theme.kt

```diff
@@ -27,20 +34,32 @@ private val LightColorScheme = lightColorScheme(
     tertiary = LightTertiary
 )
 
+@HiltViewModel
+class ThemeViewModel @Inject constructor(
+    private val settingService: SettingService
+) : ViewModel() {
+    fun getSetting(): Flow<Int> {
+        return settingService.get(Setting.ApplicationTheme)
+    }
+}
+
 @Composable
 fun CalendarTheme(
+    model: ThemeViewModel,
```

Comment:
model: ThemeViewModel = hiltViewModel() to automatically inject the viewmodel (therefore could also remove ThemeViewModel mentions from Activity.kt completely)